src/routes/settings/Connections.tsx

```diff
@@ -25,13 +35,185 @@ import {
 } from "~/components";
 import { useI18n } from "~/i18n/context";
 import { useMegaStore } from "~/state/megaStore";
-import { eify } from "~/utils";
+import { createDeepSignal, eify } from "~/utils";
+
+type IntervalChooserForm = {
+    interval: "Day" | "Week" | "Month" | "Year";
+};
+
+function mapIntervalToBudgetPeriod(
+    interval: "Day" | "Week" | "Month" | "Year"
+): BudgetPeriod {
+    switch (interval) {
+        case "Day":
+            return 0;
+        case "Week":
+            return 1;
+        case "Month":
+            return 2;
+        case "Year":
+            return 3;
+    }
+}
+
+function NwcDetails(props: { profile: NwcProfile; refetch: () => void }) {
+    const i18n = useI18n();
+    const [state, _actions] = useMegaStore();
+
+    async function toggleEnabled() {
+        try {
+            await state.mutiny_wallet?.edit_nwc_profile({
+                ...props.profile,
+                enabled: !props.profile.enabled
+            });
+            props.refetch();
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    function openInNostrClient() {
+        const uri = props.profile.nwc_uri;
+        Browser.open({ url: uri });
+    }
+
+    function openInPrimal() {
+        const uri = props.profile.nwc_uri;
+        const connectString = uri.replace("nostr+walletconnect", "primal");
+        Browser.open({ url: connectString });
+    }
+
+    async function setAutoApprove(checked: boolean) {
+        try {
+            if (checked) {
+                await state.mutiny_wallet?.set_nwc_profile_budget(
+                    props.profile.index,
+                    420n,
+                    0
+                );
+            } else {
+                await state.mutiny_wallet?.set_nwc_profile_require_approval(
+                    props.profile.index
+                );
+            }
+            props.refetch();
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    async function saveNewBudget(amountSats: bigint) {
+        try {
+            await state.mutiny_wallet?.set_nwc_profile_budget(
+                props.profile.index,
+                amountSats,
```

Comment:
Not to needlessly complicate the code but since the budget is implied as a total remaining should we set the budget amount as:
`amountSats - (props.profile.budget_amount - props.profile.budget_remaining)`
By setting it to just amountSats any change will completely reset the budget.

Reply:
I think..? this would also fix the issue where changing the day also resets the budget... not 100% on this one

Reply:
I misunderstood how the set_nwc_profile_function worked in my previous comment